scripts/merge-changelogs.ts

```diff
@@ -0,0 +1,191 @@
+import { writeFile, readFile } from 'fs/promises';
+import { GlobSync } from 'glob';
+
+const validMessageTypes = [
+  'Known Issue',
+  'Compatibility Note',
+  'New Functionality',
+  'Improvement',
+  'Fixed Issue',
+  'Updated Dependencies'
+] as const;
+
+interface Message {
+  packageNames: string[];
+  commit: string;
+  type: typeof validMessageTypes[number];
+  version: string;
+  message: string;
+}
+
+function getPackageName(changelog: string): string {
+  const packageNameRegex = /^# @sap-cloud-sdk\/(.+)$/m;
+  const title = changelog.match(packageNameRegex);
+
+  if (!changelog.startsWith('# @sap-cloud-sdk/') || !title || !title[1]) {
+    throw new Error('Changelog should have package name on first line');
+  }
+
+  return title[1];
+}
+
+function splitByVersion(changelog: string): any[] {
+  // Explaination: https://regex101.com/r/wCKs2A/2
```

Comment:
These explainaitions are nice but what if they break? I think a example or test documenting the behavior.

Reply:
I think the only scenario that would realistically break the regex would be a changed format in the input document, but a test with dummy data would not notice this. 

These links contains an example, so I would not add a long example to the source file. Test files in scripts are not executed as far as I know, so I don't know where you would add a test. 

![Screen Shot 2022-05-18 at 16 17 17](https://user-images.githubusercontent.com/26565533/169063531-882cc56f-6295-42f9-9b47-12c5623e5ed8.png)